src/components/Footer.tsx

```diff
@@ -52,14 +52,18 @@ const Footer = ({ className }: FooterProps) => {
         <div className="space-y-6 xl:space-y-4">
           <div className="flex flex-wrap gap-6 text-xl font-bold xl:gap-12 xl:text-3xl">
             <Link
-              className="hidden py-2.5 xl:inline-block xl:py-6"
+              className="relative hidden py-2.5 transition duration-300 ease-out hover:after:absolute hover:after:bottom-0 hover:after:left-0 hover:after:block hover:after:w-full hover:after:border-b-2 hover:after:border-primary hover:after:bg-primary xl:inline-block xl:py-6"
```

Comment:
1. hover動畫顯示不正常，沒有淡入效果，要記得加`opacity`
2. `transition` 也一樣要寫在after才有作用
3. 這點你可以自己決定要不要把`footer menu`，改成一個component，方便以後修改。當然不改也可以。

```suggestion
              className="relative hidden py-2.5 after:absolute after:bottom-0 after:left-0 after:block after:w-full after:border-b-2 after:border-primary after:bg-primary after:opacity-0 after:transition-opacity after:duration-300 after:ease-out hover:after:opacity-100 xl:inline-block xl:py-6"
```